File: java/eyes.selenium.common/src/main/java/com/applitools/eyes/selenium/fluent/AccessibilityRegionBySelector.java
package com.applitools.eyes.selenium.fluent;

import com.applitools.eyes.AccessibilityRegionType;
import com.applitools.eyes.fluent.GetRegion;
import com.applitools.eyes.fluent.IGetAccessibilityRegionType;
import com.applitools.eyes.serializers.BySerializer;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.openqa.selenium.By;

public class AccessibilityRegionBySelector implements GetRegion, IGetAccessibilityRegionType {

    private final AccessibilityRegionType regionType;
    @JsonSerialize(using = BySerializer.class)
    private final By selector;

    public AccessibilityRegionBySelector(By selector, AccessibilityRegionType regionType) {
        this.selector = selector;
        this.regionType = regionType;
        //selector.criteria
    }

    @Override
    public AccessibilityRegionType getAccessibilityRegionType() {
        return this.regionType;
    }

    public By getSelector() {
        return selector;
    }
}

File: dotnet/Eyes.Selenium4.DotNet/Fluent/AccessibilityRegionBySelector.cs
using OpenQA.Selenium;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Applitools.Selenium.Fluent
{
    internal class AccessibilityRegionBySelector : IGetAccessibilityRegion, IGetSeleniumRegion, IGetAccessibilityRegionType
    {
        private readonly AccessibilityRegionType regionType_;
        private readonly By selector_;
        private readonly SelectorTransformer selectorTransformer_;

        public AccessibilityRegionBySelector(By selector, AccessibilityRegionType regionType)
        {
            selector_ = selector;
            regionType_ = regionType;
            selectorTransformer_ = new SelectorTransformer();
        }

        AccessibilityRegionType IGetAccessibilityRegionType.AccessibilityRegionType => regionType_;

        public TAccessibilityRegion ToRegion()
        {
            return new SelectorAccessibilityRegion
            {
                Type = regionType_,
                Region = selectorTransformer_.GetRegionSelector(selector_)
            };
        }

        IList<IWebElement> IGetSeleniumRegion.GetElements(IWebDriver driver)
        {
            ReadOnlyCollection<IWebElement> elements = driver.FindElements(selector_);
            return elements;
        }
    }
}